### ISSUE TITLE
decide what to do with new flag that will be set when reference has issue in project

### ISSUE THREAD

**Issue by heejaechang** [dotnet/roslyn#5902]
we have many related issues where a project has invalid/missing references and our diagnostics are kind of all wrong.

we could say diagnostic itself is correct since it is showing the result of the missing/invalid references but end user experience is quite bad.

to fix this issue, one of thing ide team has done is introducing a flag that will let us know when that situation has happened (previously, there was even no way to know that has happened)

now with this new flag, we have some design decision to make.

1. when this situation happens, what should we do with user diagnostic provider/fixer.
2. where we shutdown analysis, engine itself or each features. if we shutdown in the engine itself, all consumers of the engine will never know such diagnostics ever produced. meaning no squiggle, no LB, no error list and etc but there is no easy way for user to know why this is happening.
3. how should we do de-duplication between IDE and build. the situation where this flag is set probably only happen in IDE, but most likely not happen in build, so de-duplicating diagnostics from these two sources wouldn't be that straight-forward.

there could be more thing to consider but these are what I know so far.

**Comment by srivatsn** [comment]
We discussed this and here's the conclusion:
When the flag is set for a project, it could be because the project is completely broken (like in the case of a nuget-restore-hasn't-happened-yet scenario) or it could be because one single, possibly even unnecessary reference, was missing. In the former case, turning off diagnostics makes sense but in the latter case that would be a bad experience. Moreover turning off diagnostics entirely for a project means that codefixes won't work. Ideally, if we can detect the scenarios where we know the project's broken, then we should turn off diagnostics. The only such case we know today is the nuget scenario. So we decided to do the following two things:

- The project system will detect when nuget restore hasn't happened yet and will mark all affected projects with the flag and this will be only scenario where the flag gets set.
- When the flag is set for a project, we turn off diagnostics for that project except for open files (so that we can get codefixes in those open files). 

Tagging @jasonmalinowski @Pilchie to take care of the first workitem.
